### ISSUE TITLE
Filtering in any collection crush the app after navigating to another colleciton

### ISSUE THREAD

**Issue by chelaru-artur** [payloadcms/payload#8198]
### Link to reproduction

any blank project

### Environment Info

```text
Payload: 3.0.0-beta.104
next: 15.0.0-canary.104
node: 20.17.0
```

### Describe the Bug

Previous issue with this problem was closed but bug remains: [7431](https://github.com/payloadcms/payload/issues/7431)

https://github.com/user-attachments/assets/b5cee1e3-fc63-4f16-8bf0-361d54b8cc01

### Reproduction Steps

1. Go to collection No. 1
2. Apply filter to any field
3. Go to collection No. 2
4. Apply filter to any field
5. Go to collection No. 1

https://github.com/user-attachments/assets/715ac9f5-3801-4acb-acd3-ca15d2ebc321

**Comment by paulpopus** [comment]
I haven't been able to reproduce this with those steps. For me, no matter how I navigate or click, the filters are cleared out. Any specific database when encountering this?

**Comment by chelaru-artur** [comment]
@paulpopus I'm using MongoDB. Windows OS.

**Comment by eznagovan-xmtrading** [comment]
I have same issue on Mac @paulpopus

**Comment by DanRibbens** [comment]
I was able to reproduce this one in our community test suite locally.
- Click Users in the nav
- filter by email, equals "some value"
- click on Posts in the nav
```
TypeError: Cannot read properties of undefined (reading 'Filter')
    at Condition (index.tsx:123:21)
```

**Comment by GermanJablo** [comment]
Fixed in #8260
